compiler/rustc_ast_pretty/src/pprust/state/expr.rs

```diff
@@ -341,10 +341,16 @@ impl<'a> State<'a> {
                 self.print_type(ty);
             }
             ast::ExprKind::Type(expr, ty) => {
-                let prec = AssocOp::Colon.precedence() as i8;
-                self.print_expr_maybe_paren(expr, prec);
-                self.word_space(":");
+                self.word("type_ascribe!(");
+                self.ibox(0);
+                self.print_expr(expr);
+
+                self.word(",");
+                self.space_if_not_bol();
                 self.print_type(ty);
+
+                self.end();
+                self.word(")");
```

Comment:
Can't we remove this now?

Reply:
We want to keep `type_ascripe` (which may should probably become a `builtin#` instead of a builtin macro soon) to continue testing the type system part of type ascription.